core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -248,6 +259,23 @@ public MetadataManager(
         verifyTypes();
 
         functionDecoder = new ResolvedFunctionDecoder(this::getType);
+
+        operatorCache = CacheBuilder.newBuilder()
+                .maximumSize(1000)
+                .build(CacheLoader.from(entry -> {
+                    String name = mangleOperatorName(entry.getKey());
+                    return resolveFunction(QualifiedName.of(name), fromTypes(entry.getValue()));
+                }));
+
+        coercionCache = CacheBuilder.newBuilder()
+                .maximumSize(1000)
```

Comment:
Do we want (or need?) to make this cache part of transaction state?

Reply:
I think it's transaction agnostic